src/include/sof/lib/pm_runtime.h

```diff
@@ -47,8 +47,30 @@ enum pm_runtime_context {
 struct pm_runtime_data {
 	spinlock_t lock;	/**< lock mechanism */
 	void *platform_data;	/**< platform specific data */
+#if CONFIG_DSP_RESIDENCY_COUNTERS
+	struct r_counters_data *r_counters; /**< diagnostic DSP residency counters */
+#endif
 };
 
+#if CONFIG_DSP_RESIDENCY_COUNTERS
+/**
+ * \brief DSP residency counters
+ * R0, R1, R2 are DSP residency counters which can be used differently
+ * based on platform implementation
+ */
+enum dsp_r_state {
+	r0_r_state = 0,
+	r1_r_state,
+	r2_r_state
```

Comment:
I need more that this, how am I or other user meant to debug or provide reliable information for others to debug. Can we give a general idea of each state. i.e. r0 = clocks full on - high perf mode,

Reply:
Added comments for each state, please check now